### PR TITLE
Added material assigment to links with wheel

### PR DESCRIPTION
**Pull request by michalpelka**
Changes to urdf importer.

### CODE REVIEW

**Comment by michalpelka** [comment]
Checked against `stalization/2210`

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 1]
```diff
@@ -28,10 +29,50 @@
 
 namespace ROS2
 {
+    bool CollidersMaker::GuessIsWheel(const AZStd::string& entityName)
+    {
+        AZStd::regex wheel_regex("(?i)wheel|(?i)wh_|(?i)(_wh)");
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp]: Has (case-insensitive) "wheel" anywhere in name, but has no "joint" anywhere.

Comment (adamdbrw) [reply Code/Source/RobotImporter/URDF/CollidersMaker.cpp]: has a collider
has a visual - in the link (URDF): visuals in entity/prefab are in sub-entity
has a continuous joint (component already)
and the name

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 2]
```diff
@@ -28,10 +29,50 @@
...
+        AZStd::smatch match;
+        if (AZStd::regex_search(entityName, match, wheel_regex))
+        {
+            AZ_Printf("WheelMaterial", "%s is a wheel\n", entityName.c_str());
+            return true;
+        }
+        AZ_Printf("WheelMaterial", "%s is NOT a wheel\n", entityName.c_str());
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp]: remove print

File: Assets/Prefabs/Materials/wheel_material.physxmaterial [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            <Class name="unsigned char" field="FrictionCombine" value="2" type="{72B9409A-7D1A-4831-9CFE-FCB3FADD3426}"/>
+            <Class name="unsigned char" field="RestitutionCombine" value="1" type="{72B9409A-7D1A-4831-9CFE-FCB3FADD3426}"/>
+            <Class name="float" field="Density" value="1200.0000000" type="{EA2C3E90-AFBE-44D4-A90D-FAAF79BAF93D}"/>
+            <Class name="Color" field="DebugColor" value="0.1088426 0.1088426 0.1088426 1.0000000" type="{7894072A-9050-4F0F-901B-34B1A0D29417}"/>
+        </Class>
+        <Class name="PhysicsLegacy::MaterialId" field="LegacyPhysicsMaterialId" version="1" type="{744CCE6C-9F69-4E2F-B950-DAB8514F870B}">
+            <Class name="AZ::Uuid" field="MaterialId" value="{8C7A6011-61C2-46B7-9BF4-8D4DD2A624F1}" type="{E152C105-A133-4D03-BBF8-3D4B2FBA3E2A}"/>
+        </Class>
+    </Class>
+</ObjectStream>
```
Comment (adamdbrw) [inline Assets/Prefabs/Materials/wheel_material.physxmaterial]: move this file out from Prefabs

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 4]
```diff
@@ -28,10 +29,50 @@
...
+    }
+
     CollidersMaker::CollidersMaker(AZStd::string modelPath)
         : m_modelPath(AZStd::move(modelPath))
         , m_stopBuildFlag(false)
     {
+        // get Wheel material from assets
+        bool assetFound = false;
+        AZ::Data::AssetInfo assetInfo;
+        AZStd::string watchDir;
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp:45]: unnecessary?

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 5]
```diff
@@ -219,11 +260,25 @@ namespace ROS2
         // TODO - solution: once Collider Component supports Cylinder Shape, switch to it from Shape Collider Component.
 
         AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        AZ_Assert(entity, "entity not exists");
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp]: more informative and consider also adding an if (for release)

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 6]
```diff
@@ -28,10 +29,50 @@
 
 namespace ROS2
 {
+    bool CollidersMaker::GuessIsWheel(const AZStd::string& entityName)
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp]: move out to Utils (could be used in vehicle dynamics), and will be for sure used in the Importer when adding VD components

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 7]
```diff
@@ -219,11 +260,25 @@ namespace ROS2
...
+
+        Physics::ColliderConfiguration colliderConfig;
+        const bool is_wheel = GuessIsWheel(entity->GetName());
+        if (is_wheel)
+        {
+            colliderConfig.m_materialSlots.SetMaterialAsset(0, m_wheelMaterial);
+        }
+        colliderConfig.m_position = URDF::TypeConversions::ConvertVector3(collision->origin.position);
+        if (!isPrimitiveShape)
+            colliderConfig.m_rotation = URDF::TypeConversions::ConvertQuaternion(collision->origin.rotation);
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp]: {
}

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 8]
```diff
@@ -219,11 +260,25 @@ namespace ROS2
         // TODO - solution: once Collider Component supports Cylinder Shape, switch to it from Shape Collider Component.
 
         AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        AZ_Assert(entity, "entity not exists");
         auto geometry = collision->geometry;
         bool isPrimitiveShape = geometry->type != urdf::Geometry::MESH;
+
+        Physics::ColliderConfiguration colliderConfig;
+        const bool is_wheel = GuessIsWheel(entity->GetName());
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp]: isWheelEntity

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 9]
```diff
@@ -219,11 +260,25 @@ namespace ROS2
...
+        {
+            colliderConfig.m_materialSlots.SetMaterialAsset(0, m_wheelMaterial);
+        }
+        colliderConfig.m_position = URDF::TypeConversions::ConvertVector3(collision->origin.position);
+        if (!isPrimitiveShape)
+            colliderConfig.m_rotation = URDF::TypeConversions::ConvertQuaternion(collision->origin.rotation);
+
         if (!isPrimitiveShape)
         {
-            entity->CreateComponent<PhysX::EditorColliderComponent>();
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp:226]: move code to subroutine. 
here, check isWheelEntity, warn that this case is not supported yet

File: Code/Source/RobotImporter/URDF/InertialsMaker.h [hunk 10]
```diff
@@ -13,6 +13,8 @@
 
 namespace ROS2
 {
+    constexpr AZ::u8 kMinimalNumPosSolv = 100;
+    constexpr AZ::u8 kMinimalNumVelSolv = 100;
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/InertialsMaker.h]: 50

**Comment by adamdbrw** [comment]
Fixes #168

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp (hunk 4 above)
Comment (michalpelka) [reply Code/Source/RobotImporter/URDF/CollidersMaker.cpp:45]: Unfortunately it is needed to provide reference to string here :(

**Comment by michalpelka** [comment]
@adamdbrw thank you for your review. I've applied all of your ideas. 
It was somehow major, so please take a look.

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 11]
```diff
@@ -9,29 +9,53 @@
...
-#include <Source/EditorShapeColliderComponent.h>
 
 namespace ROS2
 {
+
     CollidersMaker::CollidersMaker(AZStd::string modelPath)
         : m_modelPath(AZStd::move(modelPath))
         , m_stopBuildFlag(false)
     {
+        // get Wheel material from assets
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp]: Move out wheel material handling to a subroutine

File: Code/Source/RobotImporter/URDF/InertialsMaker.cpp [hunk 12]
```diff
@@ -26,6 +26,7 @@ namespace ROS2
         PhysX::EditorRigidBodyConfiguration rigidBodyConfiguration;
         rigidBodyConfiguration.m_mass = inertial->mass;
 
+        PhysX::RigidBodyConfiguration physxSpecificConfiguration;
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/InertialsMaker.cpp]: empty?

**Comment by michalpelka** [comment]
@adamdbrw applied your last comments.

**Review by adamdbrw — APPROVED** [review]

**Pull request merged**